### PR TITLE
Add deepspeed support for timm

### PR DESCRIPTION
**Pull request by syorami**
deepspeed=0.7.4, higher version would encounter information display problems

refer to https://github.com/microsoft/DeepSpeed/issues/2629

Command works like `python train_deepspeed.py dataset/cifar100 --dataset torch/CIFAR100 --dataset-download --model vit_large_patch16_224_in21k --pretrained --weight-decay 0 --lr 0.003 -b 64 --amp --pin-mem --epochs 20 --deepspeed --deepspeed_config configs/deepspeed_test.json`

### CODE REVIEW

File: train_deepspeed.py [hunk 1]
```diff
@@ -736,20 +760,25 @@ def train_one_epoch(
...
+            optimizer.zero_grad()
+            if loss_scaler is not None:
+                loss_scaler(
+                    loss, optimizer,
+                    clip_grad=args.clip_grad, clip_mode=args.clip_mode,
+                    parameters=model_parameters(model, exclude_head='agc' in args.clip_mode),
+                    create_graph=second_order)
+            else:
+                loss.backward(create_graph=second_order)
+                if args.clip_grad is not None:
```
Comment (ShenglongZ) [inline train_deepspeed.py:773]: 这个需要加到deepspeed的else里吗?

Comment (syorami) [reply train_deepspeed.py:773]: https://www.deepspeed.ai/docs/config-json/#gradient-clipping，deepspeed配置中可以设置
